Nodejs/Product/TestAdapterImpl/TestFrameworks/NodejsTestInfo.cs

```diff
@@ -1,36 +1,62 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.IO;
+using System.Text;
 
 namespace Microsoft.NodejsTools.TestAdapter.TestFrameworks
 {
     internal class NodejsTestInfo
     {
-        public NodejsTestInfo(string fullyQualifiedName)
+        public NodejsTestInfo(string fullyQualifiedName, string modulePath)
         {
             var parts = fullyQualifiedName.Split(new[] { "::" }, StringSplitOptions.None);
             if (parts.Length != 3)
             {
                 throw new ArgumentException("Invalid fully qualified test name", nameof(fullyQualifiedName));
             }
+
+            this.ModulePath = modulePath;
             this.ModulePath = parts[0];
             this.TestName = parts[1];
             this.TestFramework = parts[2];
         }
 
         public NodejsTestInfo(string modulePath, string testName, string testFramework, int line, int column)
         {
+            var moduleName = Path.GetFileNameWithoutExtension(modulePath);
+            var hash = GetHash(modulePath);
+
             this.ModulePath = modulePath;
+            this.ModuleName = $"{moduleName}[{hash}]";
```

Comment:
We do we need to uniquely identify this instance of the file by hashing its contents?

Reply:
Hi Bill. I'd like to comment on this since I was the one who originally implemented this. On TFS 2015 / 2017, there is a feature which shows how long a certain test has been failing ("Failing since" on a test run). Since the path to the file was part of the unique identifier, the same test running on two machines with different drive names (for example) were seen as two different tests, thus making this metric inaccurate. I'd be open to a more elegant fix for this issue, if there is one. Let me know if you have any other questions. Cheers

Reply:
Would the path relative to the project root solve this? (e.g. https://social.msdn.microsoft.com/Forums/vstudio/en-US/954346c8-cbe8-448c-80d0-d3fc27796e9c/how-to-convert-absolute-file-path-to-relative-path-in-c?forum=csharpgeneral )

Reply:
I changes this to instead of using a Hash now creates a new based on the relative path.